### ISSUE TITLE
Goliath boats don't protect you from lava

### ISSUE THREAD

**Issue by farie82** [ParadiseSS13/Paradise#12246]
**Issue Description**:
You burn while in the goliath boat

**What did you expect to happen**: 
To be safe

**What happened instead**:
I burned nearly to death. And my soul died a little

**Why is this bad/What are the consequences**:
It's a broken feature.

**Steps to reproduce the problem**:
1. Get a boat. (optionally an oar)
2. Push one self into the lava in the boat
3. Start burning and feel slightly sad inside.

**When did the problem start happening**:
Unsure

**Extra information**:

**Comment by Fox-McCloud** [comment]
Oof, yeah.

See, the lava checks for resistance flags instead of `burn_state`--this was just and oversight when it was implemented---I *could* fix that, directly, but it's going to cause some issues (I tested a quick fix and it allowed for a mech to be set on fire which...when it was destroyed, dropped a radio that you're normally never supposed to be able to access as it was an abstract item type).

The universal object damage PR should fix this though.
